### PR TITLE
New NGOpt taking into account microprediction results

### PR DESCRIPTION
**Pull request by teytaud**
User-suggested modification validated in https://dl.fbaipublicfiles.com/nevergrad/allxps/list.html
Warning: breaking change, because maintaining several metamodels is complicated.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested (if it applies)

Dashboard.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -512,6 +512,7 @@ def __init__(
 CMA = ParametrizedCMA().set_name("CMA", register=True)
 DiagonalCMA = ParametrizedCMA(diagonal=True).set_name("DiagonalCMA", register=True)
 FCMA = ParametrizedCMA(fcmaes=True).set_name("FCMA", register=True)
+ECMA = ParametrizedCMA(elitist=True).set_name("ECMA", register=True)
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:515]:
```suggestion
```
not needed

File: nevergrad/optimization/optimizerlib.py [hunk 2]
```diff
@@ -1328,7 +1329,7 @@ def __init__(
     ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
         if multivariate_optimizer is None:
-            multivariate_optimizer = CMA if self.dimension > 1 else OnePlusOne
+            multivariate_optimizer = ECMA if self.dimension > 1 else OnePlusOne
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:1332]:
```suggestion
            multivariate_optimizer = ParametrizedCMA(elitist=True) if self.dimension > 1 else OnePlusOne
```

File: nevergrad/optimization/test_optimizerlib.py [hunk 3]
```diff
@@ -581,6 +581,14 @@ def test_shiwa_dim1() -> None:
     assert recom.value < init
 
 
+continuous_case: tp.List[tp.Tuple[str, object, int, int, str]] = [
+    ("NGOpt", d, b, n, f"continuouschoice_d{d}_b{b}_n{n} ")
+    for d in [1, 2, 10, 100, 1000]
+    for b in [2 * d, 10 * d, 100 * d]
+    for n in [1, d, 10 * d]
+]
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:589]: can you explain this? I dont really like that information are encoded in a string

**Review by jrapin — CHANGES_REQUESTED** [review]

File: nevergrad/optimization/test_optimizerlib.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/optimization/test_optimizerlib.py:589]: I am checking that in all those cases we have something reasonable (no SQP or Cobyla if parallel, and one-shot methods when high num workers).

Comment (jrapin) [reply nevergrad/optimization/test_optimizerlib.py:589]: I feel f"continuouschoice_d{d}_b{b}_n{n} " is a bit useless here, maybe just make it clearer that it's a hack, like just naming it "#CONTINUOUS"

File: nevergrad/optimization/test_optimizerlib.py [hunk 4]
```diff
@@ -620,7 +629,16 @@ def test_ngopt_selection(
         pattern = rf".*{name} selected (?P<name>\w+?) optimizer\."
         match = re.match(pattern, caplog.text.splitlines()[-1])
         assert match is not None, f"Did not detect selection in logs: {caplog.text}"
-        assert match.group("name") == expected
+        choice = match.group("name")
+        if "continuouschoice_" in expected:
+            print(f"{expected} --> {choice}")
+            if num_workers >= budget:
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:635]:
```suggestion
        if expected != "#CONTINUOUS":
            assert choice == expected
        else:
            print(f"Continuous param={param} budget={budget} workers={num_workers} --> {choice}")
            if num_workers >= budget:
```

File: nevergrad/optimization/test_optimizerlib.py [hunk 5]
```diff
@@ -620,7 +629,16 @@ def test_ngopt_selection(
...
+                assert choice == "MetaTuneRecentering"
+            if num_workers > 1:
+                assert choice != "SQP"
+                assert choice != "Cobyla"
+            return
+        assert choice == expected
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:641]:
```suggestion
```

File: nevergrad/optimization/test_optimizerlib.py [hunk 6]
```diff
@@ -581,6 +581,14 @@ def test_shiwa_dim1() -> None:
     assert recom.value < init
 
 
+continuous_case: tp.List[tp.Tuple[str, object, int, int, str]] = [
+    ("NGOpt", d, b, n, f"continuouschoice_d{d}_b{b}_n{n} ")
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:585]:
```suggestion
    ("NGOpt", d, b, n, f"#CONTINUOUS")
```

File: nevergrad/optimization/test_optimizerlib.py [hunk 7]
```diff
@@ -581,6 +581,14 @@ def test_shiwa_dim1() -> None:
     assert recom.value < init
 
 
+continuous_case: tp.List[tp.Tuple[str, object, int, int, str]] = [
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:584]:
```suggestion
continuous_cases: tp.List[tp.Tuple[str, object, int, int, str]] = [
```

File: nevergrad/optimization/test_optimizerlib.py [hunk 8]
```diff
@@ -608,7 +616,8 @@ def test_shiwa_dim1() -> None:
         ("NGOpt", ng.p.TransitionChoice(range(3), repetitions=10), 10, 2, "DiscreteLenglerOnePlusOne"),
         ("NGO", 1, 10, 1, "Cobyla"),
         ("NGO", 1, 10, 2, "OnePlusOne"),
-    ],  # pylint: disable=too-many-arguments
+    ]
+    + continuous_case,  # pylint: disable=too-many-arguments
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:620]:
```suggestion
    + continuous_cases,  # pylint: disable=too-many-arguments
```

**Pull request closed without merging**